### DISCUSSION TITLE
nextjs on vercel / zeit writing to disk fs.writeFile in server.js

### DISCUSSION THREAD

**Discussion by pleaseshutup** [vercel/next.js#12429] [Help] [answered]
I am using a custom server with nextjs to inject some serverSideProps / getInitialProps into files. What I am doing is having something on the server update a json file every minute or so and then the getServerSideProps/getInitialProps functions read the json file from disk to populate data quickly (without having to do a network request for data).

Everything works great in localhost and when doing a build/running the build locally. In production / after deploying to vercel i get the error: Internal Server Error when trying to open a page. Vercel offers build logs but not any logs for the running custom server.

Is this likely a problem to do with unable to write to disk in the vm's that vercel uses to deploy apps?

Is there an example out there of how I can read/write to disk with a custom server in nextjs for this purpose?

**Comment by revskill10** [comment]
As i know, you can write to `/tmp` folder on Vercel (i've not tested yet). Could you retry ?

**Reply by pleaseshutup** [reply]
I changed it to writing to /tmp and it still produces the same error... hrm.
Anyone have an example of a custom server.js file writing files to disc that works in production that I can look at?

**Comment by pleaseshutup** [comment]
As an update:
I wrapped my fs.readFileSync in getServerSideProps into try/catch and was able to see on the getServerSideProps the error
Error: ENOENT: no such file or directory, open '/tmp/mydata.json'

Which likely means the server.js side isn't able to fs.writeFile('/tmp/mydata.json', JSON.stringify(data), ()=>{}); 

so my question remains the same, how can a custom server write files to disc in the vercel hosting vms?

**Comment by paulogdm** [comment, marked as answer]
It is not possible to deploy a custom server to the Vercel platform. So it is likely the file was not there since the code in the custom server is ignored.

Also, please provide a repository with some code so we can have more context.

**Reply by pleaseshutup** [reply]
Ah yes thank you for reminding me.

We can close this out. I had, after many hours of frustration finally learned that when you deploy to vercel the custom server is totally ignored.
